packages/vanilla/src/components/horizontalrule/horizontalrule.template.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Horizontal Rule template literal
+ *
+ * @param {object} props Property object passed into the template
+ * @param {string} props.style solid vs dashed [''|'dashed']
+ * @param {string} props.size size of the hr [''|'small'|'medium'|'large']
+ * @param {string} props.contrast color of the rule [''|'low'|'high']
+ * @param {string} props.weight weight of the rule [''|'thick']
+ * @returns {string} ES6 template literal of the horizontal rule
+ */
+const horizontalruleTemplate = ({ style, size, contrast, weight }) => {
+  const hrStyle = style === 'dashed' ? 'bx--hr--dashed' : '';
+  const hrSize = size && size !== '' ? `bx--hr--${size}` : '';
```

Comment:
I'm thinking `size` would fail if it equals `''`, so I don't think the additional `&& size !== ''` here is necessary. Same comment for `L14`.

```suggestion
  const hrSize = size ? `bx--hr--${size}` : '';
```

Reply:
I guess it would fail in the sense where it would add in a class name `bx--hr--`, was trying to avoid that in the output. Just checking for `size` would certainly check for a null value, but doesn't account for empty string. I guess same would be if the user puts in any garbage string though, so we can just remove to be more terse anyway.

Reply:
Added additional checks to make sure the proper classes are added.

Reply:
Hmmm, unless I'm misunderstanding something just checking for `size` will fail if `null`, `undefined`, `false`, or if it's an empty string `''`.

https://jsfiddle.net/r5ockdag/